src/anomaly/metaml.rs

```diff
@@ -1,214 +1,267 @@
+    ///
+    /// // Model must be trained before using .save()
+    /// model.train(dataset);
+    ///
+    /// // Save the model to disk for later use
+    /// let output_path = Path::new("path/to/output/model.file");
+    /// model.save(&output_path);
+    /// ```
+    fn save(&self, path: &Path) -> Result<(), String>;
+}
+
+/// Represents the training data for a MetaML model
+///
+/// # Invariants:
+/// * The number of columns in the features data is 6
+/// * The number of rows in the features data is equal to the number of rows in the target data
+/// * The data at row `i` of the features data corresponds to the data at row `i` of the targets data
+pub struct MetaMLDataset {
+    pub features: Vec<[f32; 6]>,
+    pub targets: Vec<f32>,
+}
+
+impl MetaMLDataset {
+    /// Creates a dataset for training a meta-ml model from a set of feature values
+    /// and their corresponding target values
+    ///
+    /// # Errors:
+    /// * If the number of columns in the features data isn't 6
+    /// * If the number of rows in the features data doesn't match the number of
+    /// elements in the targets data
+    ///
+    /// # Examples
+    /// ```
+    /// # use ndarray::{Array1, Array2, arr1, arr2};
+    /// # use clam::anomaly::metaml::MetaMLDataset;
+    /// #
+    /// # let features: Vec<[f32; 6]> = vec![[0.0, 0.0, 0.0, 0.0, 0.0, 0.0]];
+    /// # let targets: Vec<f32> = vec![0.0];
+    ///
+    /// // Remember features must have the same number of rows as targets
+    /// assert!(features.len() == targets.len());
+    ///
+    /// let dataset: MetaMLDataset = MetaMLDataset::new(features, targets).unwrap();
+    ///
+    /// // Do something with the dataset
+    /// ```
+    pub fn new(features: Vec<[f32; 6]>, targets: Vec<f32>) -> Result<Self, String> {
+        // TODO better error checking once the rust branch is merged into master
+        if features.len() != targets.len() {
+            Err("Different number of features and targets in input data".to_string())
+        } else {
+            Ok(MetaMLDataset { features, targets })
+        }
+    }
+
+    /// Creates a dataset for training a meta-ml model from input data on disk.
+    ///
+    /// # Panics
+    /// * If either of the given paths can't be converted to a string
+    /// * If either of the given files can't be found, opened, or parsed as `f32`s
+    /// * If the data contained within the features file isn't two-dimensional
+    /// * If the data contained within the targets file isn't one-dimensional
+    /// * If the number of columns in the features data isn't 6
+    /// * If the number of rows in the features data doesn't match the number of
+    /// elements in the targets data
+    ///
+    /// # Examples
+    /// ```no_run
+    /// # use clam::anomaly::metaml::MetaMLDataset;
+    /// # use std::path::Path;
+    /// #
+    /// // File paths to training data in the numpy .npy file format
+    /// let features_path = Path::new("path/to/data/features.npy");
```

Comment:
Can we add some data to this repo so that snippets like this could be run? See the data folder for examples